src/components/stories/Stories.jsx

```diff
@@ -1,20 +1,134 @@
+      if (error) {
+        loading = false;
+        alert("Can not create the story! Please try again");
+      } else if (data) {
+        // clear data
+        loading = false;
+        reset({
+          title: "",
+          desc: "",
+        });
+        setCountry("");
+        setRegion("");
+        // close modal
+        handleClose();
+      }
+    }
+  };
+
   return (
     <div className="stories">
       <div className="main_title">
         <h1>User Story</h1>
-        <Button className="story_btn" as={Link} to={"/404/"}>
+        <Button className="story_btn" onClick={handleShow}>
           Create your Story
         </Button>
       </div>
       <Row className="mb-2 w-100 petList">
         {userData && userData.map((u) => <UserCard key={u.id} userData={u} />)}
       </Row>
+
+      <Modal show={show} onHide={handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Create Your Story</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div className="formContainer">
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <label>Title</label>
+              <input {...register("title", { required: true })} />
+              {errors.title?.type === "required" && (
+                <p className="errMsg">*Title is required!!!</p>
+              )}
+              <label>Description</label>
+              <textarea type="text" {...register("desc", { required: true })} />
+              {errors.desc?.type === "required" && (
+                <p className="errMsg">*Description is required!!!</p>
+              )}
+              <label>Location</label>
+              <div>
+                <CountryDropdown
+                  className="country"
+                  name="country"
+                  value={country}
+                  onChange={(val) => setCountry(val)}
+                  required
+                />
+                <RegionDropdown
+                  className="region"
+                  name="region"
+                  blankOptionLabel="No Country Selected"
+                  country={country}
+                  value={region}
+                  onChange={(val) => setRegion(val)}
+                  required
+                />
+              </div>
+              <br />
+              <hr />
+              <div className="btnContainer">
+                <button
+                  className="mBtn clearBtn"
+                  onClick={() => {
+                    reset({
+                      title: "",
+                      desc: "",
+                    });
+                    setCountry("");
+                    setRegion("");
+                  }}
+                >
+                  Clear
+                </button>
+                <button className="mBtn closeBtn" onClick={handleClose}>
+                  Close
+                </button>
+                <button className="mBtn saveBtn" type="submit">
+                  Save
+                </button>
```

Comment:
if you want you could use `    <FetchingButton fetching={fetching} action="Save" className="styles here" />`

fetching is the same thing as loading. got it from the library `react-supabase`

for example `const [{ count, data, error, fetching }, execute] = useInsert('todos')`